src/main/java/Duke.java

```diff
@@ -1,10 +1,251 @@
+    }
+
+    /**
+     * Marks tasks in list[] as done, if they exist or has not been completed.
+     *
+     * @param taskNumber Number n associated with the task, where n is the nth task in list.
+     */
+    public static void markTaskAsDone(int taskNumber) {
+        boolean isExists = taskNumber >= 0 && taskNumber < taskCount;
+
+        printTopLine();
+        if (!isExists) {
+            System.out.println("    │ Wha- Hey! Task does not exist!");
+        } else if (list[taskNumber].isDone) {
+            System.out.println("    │ Dude... you've done the task already.");
+        } else if (isExists){
+            taskCompleted++;
+            list[taskNumber].isDone = true;
+            System.out.println("    │ About time. I've mark that task as done:");
+            System.out.println("    │   [" + list[taskNumber].getStatusIcon() + "]" + list[taskNumber].description);
+        }
+        printBottomLine();
+    }
+
+    /**
+     * Checks whether there exist an input after user input the command word.
+     *
+     * @param input Expected input from user after the command word taskType is typed.
+     * @return true if input is empty, false if input is not empty.
+     */
+    public static boolean isEmpty(String input) {
+        return input.equals("");
+    }
+
+
+    /**
+     * Prints greeting message to user when code is ran.
+     */
+    public static void greetUser() {
+        printLogo();
+        printTopLine();
+        System.out.println(INDENT + "*Sigh* Hi... I'm Tired                                             │\n"
+                + INDENT + "What do you want from me?                                          │");
+        printBottomLine();
+    }
+
+    /**
+     * Prints a snarky remark to user.
+     */
+    public static void mockUser() {
+        printTopLine();
+        System.out.println(INDENT + "You know what a todo list bot is?\n"
+                + INDENT + "I'm a todo list bot. So stop chatting with me.");
+        printBottomLine();
+    }
+
+    /**
+     * Prints farewell message to user and exits code.
+     */
+    public static void byeUser() {
+        printTopLine();
+        System.out.println(INDENT + "\"Only in the agony of parting do we look into the depths of love.\" │\n"
+                + INDENT + " —— George Eliot                                                   │\n"
+                + INDENT + "                                                                   │\n"
+                + INDENT + "Ha! As if I care! Goodbye!!                                        │");
+        printBottomLine();
+    }
+
+    /**
+     * Engages user base on what the user has typed and executes a corresponding command.
+     */
+    public static void engageUser() {
+        Scanner text = new Scanner(System.in);
+        String taskType;
+        String taskName;
+        String taskDetails = "";
+
+        String userInput;
+        String[] parts;
+        boolean isUsing = true;
```

Comment:
Consider having these names be more [self-explanatory](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-name-to-explain).
```suggestion
        String[] words;
        boolean isExit = false;
```